XCode Build Error: Hub module missing OrderedCollections dependency

## Summary
The `Hub` module from `swift-transformers` (a dependency of WhisperKit) fails to link because it's missing an explicit dependency on `OrderedCollections`, even though it uses `OrderedCollections.OrderedDictionary`.

## Error Message
```
Undefined symbols for architecture arm64:
  "OrderedCollections.OrderedDictionary.init<A where A1: Swift.Sequence, A1.Element == (key: A, value: B)>(uniqueKeysWithValues: A1) -> OrderedCollections.OrderedDictionary<A, B>", referenced from:
      Hub.Config.jinjaValue() -> Jinja.Value in Config.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1
```

## Build Warnings
During the build, multiple warnings appear indicating missing dependencies:
```
warning: 'Hub' is missing a dependency on 'OrderedCollections' because dependency scan of Swift module 'Hub' discovered a dependency on 'OrderedCollections'
warning: 'Hub' is missing a dependency on 'InternalCollectionsUtilities' because dependency scan of Swift module 'Hub' discovered a dependency on 'InternalCollectionsUtilities'
warning: 'Hub' is missing a dependency on 'Jinja' because dependency scan of Swift module 'Hub' discovered a dependency on 'Jinja'
```

## Environment
- **Xcode Version**: 26.1.1 (Build 17B100)
- **Platform**: iOS Simulator (arm64)
- **SDK**: iphonesimulator26.1
- **macOS**: macOS 26.1.0 (Sequoia)

## Package Versions
```json
{
  "packageDependencies": {
    "whisperkit": "https://github.com/argmaxinc/whisperkit @ main (cb00a08)",
    "swift-transformers": "https://github.com/huggingface/swift-transformers.git @ 1.1.2",
    "swift-collections": "https://github.com/apple/swift-collections.git @ 1.3.0",
    "Jinja": "https://github.com/huggingface/swift-jinja.git @ 2.1.1"
  }
}
```

## Steps to Reproduce
1. Add WhisperKit as a Swift Package dependency in Xcode
2. Build the project for iOS Simulator (arm64 architecture)
3. Build fails during linking of the `Hub` target from `swift-transformers`

--- a/Package.swift
+++ b/Package.swift
@@ -17,11 +17,22 @@ let package = Package(
         .library(name: "Transformers", targets: ["Tokenizers", "Generation", "Models"]),
     ],
     dependencies: [
-        .package(url: "https://github.com/huggingface/swift-jinja.git", from: "2.0.0")
+        .package(url: "https://github.com/huggingface/swift-jinja.git", from: "2.0.0"),
+        .package(url: "https://github.com/apple/swift-collections.git", from: "1.0.0"),
     ],
     targets: [
         .target(name: "Generation", dependencies: ["Tokenizers"]),
-        .target(name: "Hub", dependencies: [.product(name: "Jinja", package: "swift-jinja")], resources: [.process("Resources")], swiftSettings: swiftSettings),
+        .target(
+            name: "Hub",
+            dependencies: [
+                .product(name: "Jinja", package: "swift-jinja"),
+                .product(name: "OrderedCollections", package: "swift-collections"),
+            ],
+            resources: [
+                .process("Resources")
+            ],
+            swiftSettings: swiftSettings
+        ),
         .target(name: "Models", dependencies: ["Tokenizers", "Generation"]),
         .target(name: "Tokenizers", dependencies: ["Hub", .product(name: "Jinja", package: "swift-jinja")]),
         .testTarget(name: "GenerationTests", dependencies: ["Generation"]),

